src/Storages/ObjectStorage/HDFS/ReadBufferFromHDFS.cpp

```diff
@@ -197,7 +197,7 @@ ReadBufferFromHDFS::ReadBufferFromHDFS(
         size_t read_until_position_,
         bool use_external_buffer_,
         std::optional<size_t> file_size_)
-    : ReadBufferFromFileBase(read_settings_.remote_fs_buffer_size, nullptr, 0)
+    : ReadBufferFromFileBase()
```

Comment:
@CheSema with this fix, when `use_external_buffer` is true, Who is responsible for allocating the memory of `ReadBufferFromHDFS`

Reply:
Agree with @baibaichen. When `use_external_buffer ` is true, the program shoud crash.

Reply:
Is there no test for hdfs now?

Reply:
@CheSema We found an issue in apache gluten after creating `ReadBufferFromHDFS` with `use_external_buffer` is true. Maybe we shouldn't modify `ReadBufferFromHDFS`. Before the change, when `use_external_buffer ` is true,  `ReadBufferFromHDFS` would allocate 1 buffer internal instead of 2 buffers.

Reply:
I'm reverting this change. Thanks for pointing out. It seems no tests for it at all.

But the question is really actual 
> when use_external_buffer is true, Who is responsible for allocating the memory of ReadBufferFromHDFS?
Before this change, `ReadBufferFromHDFS` allocates memory regardless the `use_external_buffer=1` parameter. This is not logical.